plugins/parquet/parquetembed.cpp

```diff
@@ -0,0 +1,1982 @@
+    if (row->GetType() == rapidjson::kObjectType)
+        row->AddMember(key, value, jsonAlloc);
+    else
+        row->PushBack(value, jsonAlloc);
+}
+
+/**
+ * @brief Writes the value to the parquet file using the StreamWriter from the ParquetHelper class.
+ *
+ * @param len Number of chars in value.
+ * @param value pointer to value of parameter.
+ * @param field RtlFieldInfo holds meta information about the embed context.
+ * @param r_parquet Shared pointer to helper class that operates the parquet functions for us.
+ */
+void bindUtf8Param(unsigned len, const char *value, const RtlFieldInfo *field, std::shared_ptr<ParquetHelper> r_parquet)
+{
+    rapidjson::Value key = rapidjson::Value(field->name, jsonAlloc);
+    rapidjson::Value val = rapidjson::Value(value, rtlUtf8Size(len, value), jsonAlloc);
+
+    addMember(r_parquet, key, val);
+}
+
+/**
+ * @brief Writes the value to the parquet file using the StreamWriter from the ParquetHelper class.
+ *
+ * @param len Number of chars in value.
+ * @param value pointer to value of parameter.
+ * @param field RtlFieldInfo holds meta information about the embed context.
+ * @param r_parquet Shared pointer to helper class that operates the parquet functions for us.
+ */
+void bindStringParam(unsigned len, const char *value, const RtlFieldInfo *field, std::shared_ptr<ParquetHelper> r_parquet)
+{
+    size32_t utf8chars;
+    rtlDataAttr utf8;
+    rtlStrToUtf8X(utf8chars, utf8.refstr(), len, value);
+
+    rapidjson::Value key = rapidjson::Value(field->name, jsonAlloc);
+    rapidjson::Value val = rapidjson::Value(std::string(utf8.getstr(), rtlUtf8Size(utf8chars, utf8.getdata())), jsonAlloc);
+
+    addMember(r_parquet, key, val);
+}
+
+/**
+ * @brief Writes the value to the parquet file using the StreamWriter from the ParquetHelper class.
+ *
+ * @param value pointer to value of parameter.
+ * @param field RtlFieldInfo holds meta information about the embed context.
+ * @param r_parquet Shared pointer to helper class that operates the parquet functions for us.
+ */
+void bindBoolParam(bool value, const RtlFieldInfo *field, std::shared_ptr<ParquetHelper> r_parquet)
+{
+    rapidjson::Value key = rapidjson::Value(field->name, jsonAlloc);
+    rapidjson::Value val = rapidjson::Value(value);
+
+    addMember(r_parquet, key, val);
+}
+
+/**
+ * @brief Writes the value to the parquet file using the StreamWriter from the ParquetHelper class.
+ *
+ * @param len Number of chars in value.
+ * @param value pointer to value of parameter.
+ * @param field RtlFieldInfo holds meta information about the embed context.
+ * @param r_parquet Shared pointer to helper class that operates the parquet functions for us.
+ */
+void bindDataParam(unsigned len, const char *value, const RtlFieldInfo *field, std::shared_ptr<ParquetHelper> r_parquet)
+{
+    size32_t utf8chars;
+    rtlDataAttr data;
+    rtlStrToUtf8X(utf8chars, data.refstr(), len, value);
+    size32_t utf8len = rtlUtf8Size(utf8chars, data.getstr());
```

Comment:
Is this correct that it is going via utf8?  I thought it mapped to a large binary.  I think this will cause data corruption/invalid results.

Reply:
Removed UTF-8 conversion.